lib/Vend/Tax/Avalara.pm

```diff
@@ -0,0 +1,1343 @@
+
+Location city used to determine jurisdiction of merchant.
+
+=item B<nexus_state> - default: __NEXUS_STATE__
+
+Location state used to determine jurisdiction of merchant.
+
+=item B<nexus_zip> - default: __NEXUS_ZIP__
+
+Location zip used to determine jurisdiction of merchant.
+
+=item B<nexus_country> - default: __NEXUS_COUNTRY__
+
+Location country used to determine jurisdiction of merchant.
+
+=item B<no_nexus_required> - default: undef
+
+When Perly true, allows the post_transaction() routine called from
+[send-tax-transaction] to create the tax transaction regardless of nexus for
+the order's shipping jurisdiction. Allows for accounting with all orders in
+Avalara instead of only those orders that happen to collect tax.
+
+=item B<invoice_number> = default: undef
+
+Invoice number applied to tax transaction created in [send-tax-transaction].
+Typically would be set to the Interchange order number (and is set accordingly
+in send_tax_transaction job).
+
+=back
+
+=cut
+
+### accessors
+{
+    my @acc = (
+        json => q{JSON->new->pretty->allow_nonref->canonical},
+        ua => q{LWP::UserAgent->new},
+        url => q{undef},
+
+        user => q{$::Variable->{AVALARA_USER}},
+        password => q{$::Variable->{AVALARA_PASSWORD}},
+        _basic => q{encode_base64(sprintf ('%s:%s', $self->user, $self->password))},
+
+        product_tax_code_field => q{'product_tax_code'},
+        shipping_tax_code => q{'FR'},
+        handling_tax_code => q{'OH010000'},
+
+        customer => q{$::Session->{username} || 'NONE'},
+        transaction_date => q{$self->tag->time({},'%F')},
+
+        nexus => q{{}},
+        load_line_items => q{undef},
+        sent_field => q{'tax_sent'},
+        sent_success_value => q{1},
+        invoice_number => q{undef},
+        no_nexus_required => q{undef},
+
+        table => q{$::Variable->{TAXAVERAGES_TABLE} || 'tax_averages'},
+
+        nexus_address => q{$::Variable->{NEXUS_ADDRESS}},
+        nexus_city => q{$::Variable->{NEXUS_CITY}},
+        nexus_state => q{$::Variable->{NEXUS_STATE}},
+        nexus_zip => q{$::Variable->{NEXUS_ZIP}},
+        nexus_country => q{$::Variable->{NEXUS_COUNTRY}},
+
+        http_response_stack => q{[]},
+    );
+
+    sub summary_rates {
+        my $self = shift;
+        my $p = \$self->{_summary_rates};
+        if (@_) {
+            my $v = shift;
+            die "summary_rates value '$v' not an ARRAY"
+                unless $self->_reftyped($v) eq 'ARRAY';
+            $$p = $v;
+        }
+        $$p ||= [];
+        return @{$$p} if wantarray;
+        return $$p;
+    }
+
+    # Overriding zip since Avalara will only work properly with 5-digit zips
+    sub zip {
+        my $self = shift;
+        local $_ = $self->SUPER::zip(@_);
+
+        return $_ if !defined
+            or $self->country ne 'US'
+            or /^\d{5}$/;
+
+        my $v = $_;
+        s/\D+//g;
+        $_ = substr ($_, 0, 5);
+        /^\d{5}$/ or return $v;
```

Comment:
Would probably be good to use regex option `/a` here, since you only want to match ASCII digits and the default nowadays is to match all sorts of odd Unicode digits.

Reply:
Adjusted.